src/controller/proxy/controller.go

```diff
@@ -209,6 +236,19 @@ func (c *controller) putBlobToLocal(remoteRepo string, localRepo string, desc di
 }
 
 func (c *controller) waitAndPushManifest(ctx context.Context, remoteRepo string, man distribution.Manifest, art lib.ArtifactInfo, contType string, r RemoteInterface) {
+	if contType == manifestlist.MediaTypeManifestList {
+		_, payload, err := man.Payload()
+		if err != nil {
+			log.Errorf("failed to get payload, error %v", err)
+		}
+		dig := digest.FromBytes([]byte(payload))
```

Comment:
This digest will be used to compare with the digest in the header in the response from remote.

It will be broken if the remote site, such as dockerhub use a different way to calculate the digest.

How about using FetchOrSave when fetching the cache, so the digest is the value from the response?

Reply:
Updated